libs/langchain/langchain/tools/graphql/tool.py

```diff
@@ -13,12 +13,17 @@ class BaseGraphQLTool(BaseTool):
 
     name = "query_graphql"
     description = """\
-    Input to this tool is a detailed and correct GraphQL query, output is a result from the API.
+    Input to this tool is a detailed and correct GraphQL query and optional query variables, graphql endpoint and request headers whereas output is the json format string of the result from the API.
     If the query is not correct, an error message will be returned.
     If an error is returned with 'Bad request' in it, rewrite the query and try again.
     If an error is returned with 'Unauthorized' in it, do not try again, but tell the user to change their authentication.
 
-    Example Input: query {{ allUsers {{ id, name, email }} }}\
+    Example Input: 
+    query: 'query($name: String) {{ allUsers(name: $name) {{ id, name, email }} }}'
+    query_variables: '{{"name": "John Wan"}}'
+    graphql_endpoint: 'http://testserver/graphql'
+    headers: '{{"Authorization": "Bearer testtoken", "Accept": "application/json"}}'
```

Comment:
Why is the LLM asked to output headers including auth? What's a good use case that explain in which situations we'd want the LLM to be able to fill in auth?